tests/test_tron_tools.py

```diff
@@ -559,3 +559,22 @@ def test_get_tron_namespaces_for_cluster_default(self, mock_system_tron_config,
             soa_dir=soa_dir,
         )
         assert namespaces == expected_namespaces
+
+    @mock.patch('os.walk', autospec=True)
+    @mock.patch('os.listdir', autospec=True)
+    def test_get_tron_namespaces_for_cluster_conflict(self, mock_ls, mock_walk):
+        cluster_name = 'stage'
+        mock_walk.return_value = [
+            ('/my_soa_dir/cool', [], ['tron-stage.yaml']),
+        ]
+        mock_ls.return_value = ['cool.yaml']
+        soa_dir = '/my_soa_dir'
+
+        try:
+            tron_tools.get_tron_namespaces_for_cluster(
+                cluster=cluster_name,
+                soa_dir=soa_dir,
+            )
+            assert False, "no exception raised on conflicting namespaces"
+        except tron_tools.ConflictingNamespacesError as e:
```

Comment:
`pytest.raises` can do this assertion for you. Should fix the test coverage issue in pre-commit.